### PR TITLE
PIL-849 PIL-850 Uniswap data points

### PR DESCRIPTION
**Pull request by graszka22**

### CODE REVIEW

**Review by IAmKio — COMMENTED** [review]
Could we get some tests around this please?

**Comment by graszka22** [comment]
@IAmKio the tests are with some dirty mocks, I'm not sure if it's possible to test these functions without them 😅

File: src/constants/liquidityPoolsConstants.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+export const UPDATE_POOL_DATA = 'UPDATE_POOL_DATA';
+export const SET_FETCHING_POOL_DATA = 'SET_FETCHING_POOL_DATA';
+
+export const UNISWAP_PLR_ETH_POOL_ADDRESS = '0xae2d4004241254aed3f93873604d39883c8259f0';
```
Comment (poocart) [inline src/constants/liquidityPoolsConstants.js:24]: Maybe we can deploy and add one for Kovan?

File: src/services/sablier.js [hunk 2]
```diff
@@ -283,21 +223,8 @@ export const getCreateStreamFeeAndTransaction = async (
...
-      from: sender,
-      to: asset.address,
-      data: approveTransactionData,
-      amount: 0,
-      symbol: ETH,
-      sequentialSmartWalletTransactions: [sablierCreateStreamTransaction],
-    };
-  }
+  sablierCreateStreamTransaction = await addAllowanceTransaction(
+    sablierCreateStreamTransaction, sender, getEnv().SABLIER_CONTRACT_ADDRESS, asset, amount);
```
Comment (poocart) [inline src/services/sablier.js:227]: The transaction will fail if the required amount is already allowed.

File: src/services/uniswap.js [hunk 3]
```diff
@@ -344,3 +346,96 @@ export const fetchUniswapSupportedTokens = async (supportedAssetCodes: string[])
...
+  let addLiquidityTransaction = {
+    from: sender,
+    to: ADDRESSES.router,
+    data: addLiquidityTransactionData,
+    amount: utils.formatUnits(ethAmount, 18),
+    symbol: ETH,
+  };
+
+  addLiquidityTransaction = await addAllowanceTransaction(
+    addLiquidityTransaction, sender, ADDRESSES.router, tokenAsset, tokenAmount);
```
Comment (poocart) [inline src/services/uniswap.js:406]: The transaction will fail if amount is already allowed.

File: src/constants/liquidityPoolsConstants.js (hunk 1 above)
Comment (graszka22) [reply src/constants/liquidityPoolsConstants.js]: it's not used anywhere, the uniswap router will figure out the pool address himself on both kovan and mainnet, removing 👍

File: src/services/sablier.js (hunk 2 above)
Comment (graszka22) [reply src/services/sablier.js:227]: The function `addAllowanceTransaction` checks for allowance and adds allowance transaction only when it's needed
And btw why would it fail? Afaik you can ask for allowance multiple times for whatever amount you want.

Comment (poocart) [reply src/services/sablier.js:227]: The estimation will fail and then transaction.

**Pull request merged**